### ISSUE TITLE
Using top-level config file options causes `KeyError` exception

### ISSUE THREAD

**Issue by maxrake** [ewels/rich-codex#35]
Using a config file with some top-level options causes a `KeyError` exception.

For example, using this configuration, saved as `.rich-codex.yml`:

```yaml
---
# Any top-level config option in `self.class_config_attrs`
# will cause an exception. This one is in that list.
terminal_width: 88
# This one is not in that list and configs that only
# contain `outputs` will not hit the exception.
outputs:
  - command: "python -m pip list"
    fake_command: "pip list"
    title: Installed Packages
    img_paths:
      - docs/img/pip_list.svg
```

...causes this output:

```python 
❯ rich-codex --skip-git-checks
INFO     rich-codex ⚡️📖⚡️ version 1.2.4
INFO     Found 1 config file
Traceback (most recent call last):
  File "/Users/maxrake/dev/delme/rich-codex/.venv/bin/rich-codex", line 8, in <module>
    sys.exit(main())
  File "/Users/maxrake/dev/delme/rich-codex/.venv/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/maxrake/dev/delme/rich-codex/.venv/lib/python3.10/site-packages/rich_click/rich_command.py", line 19, in main
    rv = super().main(*args, standalone_mode=False, **kwargs)
  File "/Users/maxrake/dev/delme/rich-codex/.venv/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/maxrake/dev/delme/rich-codex/.venv/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/maxrake/dev/delme/rich-codex/.venv/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/maxrake/dev/delme/rich-codex/src/rich_codex/cli.py", line 441, in main
    codex_obj.parse_configs()
  File "/Users/maxrake/dev/delme/rich-codex/src/rich_codex/codex_search.py", line 312, in parse_configs
    self.parse_config(config, yaml.safe_load(fh))
  File "/Users/maxrake/dev/delme/rich-codex/src/rich_codex/codex_search.py", line 321, in parse_config
    setattr(self, cls, config["cls"])
KeyError: 'cls'
```

**Comment by ewels** [comment]
Many thanks for these issues and fixes 🙌🏻  I really appreciate the time put into explaining the error and fix ⭐

**Comment by ewels** [comment]
Released in v1.2.5 - https://github.com/ewels/rich-codex/releases/tag/v1.2.5
